internal/db/schema/manager.go

```diff
@@ -156,22 +156,32 @@ func (b *Manager) RollForward(ctx context.Context) error {
 		return errors.New(errors.NotSpecificIntegrity, op, fmt.Sprintf("schema is dirty with version %d", curVersion))
 	}
 
-	return b.runMigrations(ctx, newStatementProvider(b.dialect, curVersion))
+	if err = b.runMigrations(ctx, newStatementProvider(b.dialect, curVersion)); err != nil {
+		return errors.Wrap(err, op)
+	}
+	return nil
 }
 
 // runMigrations passes migration queries to a database driver and manages
 // the version and dirty bit.  Cancelation or deadline/timeout is managed
 // through the passed in context.
-func (b *Manager) runMigrations(ctx context.Context, qp *statementProvider) error {
+func (b *Manager) runMigrations(ctx context.Context, qp *statementProvider) (err error) {
 	const op = "schema.(Manager).runMigrations"
 
 	if err := b.driver.StartRun(ctx); err != nil {
-		return err
+		return errors.Wrap(err, op)
+	}
+	if err := b.driver.EnsureVersionTable(ctx); err != nil {
+		return errors.Wrap(err, op)
 	}
 	for qp.Next() {
 		select {
 		case <-ctx.Done():
-			return errors.Wrap(ctx.Err(), op)
+			err := ctx.Err()
+			if commitErr := b.driver.CommitRun(); commitErr != nil {
```

Comment:
I'm not sure if this is the right thing to do if we have timed out.  I didn't really want to expose a rollback interface since it wouldn't be usable by non-transaction based drivers. @mgaffney

Reply:
Committing the transaction if there is a time out seems like a bad option. You could add a `Rollbacker` interface with `Rollback` being the only method and then call it here if the driver supports it.